Don't panic when --docker-run is combined with --name

thallgren opened the pull request:
## Description

The code that parses arguments after `--` would panic on flags that
used `--flag value` instead of `--flag=value`. This commit ensures that
both variants work.

## Checklist

 - [x] I made sure to update `./CHANGELOG.md`.
 - [ ] I made sure to add any docs changes required for my change (including release notes).
 - [x] My change is adequately tested.
 - [ ] I updated `DEVELOPING.md` with any any special dev tricks I had to use to work on this code efficiently.
 - [ ] I updated `TELEMETRY.md` if I added, changed, or removed a metric name.
 - [ ] Once my PR is ready to have integration tests ran, I posted the PR in #telepresence-dev in the datawire-oss slack so that the "ok to test" label can be applied.

Commit:
Don't panic when `--docker-run` is combined with `--name <name>`

The code that parses arguments after `--` would panic on flags that
used `--flag value` instead of `--flag=value`. This commit ensures that
both variants work.

Closes #2593

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

Commit:
Add some more ticket/pull links to the CHANGELOG

Signed-off-by: Thomas Hallgren <thomas@datawire.io>

josecv commented on pkg/client/cli/intercept/state.go:
```diff
@@ -329,34 +329,43 @@ func (s *state) CreateRequest(ctx context.Context) (*connector.CreateInterceptRe
 	return ir, nil
 }
 
+func getArg(args []string, flag string) (string, error) {
```
maybe this should be called `getArgForDocker`?

josecv commented on pkg/client/cli/intercept/state.go:
```diff
@@ -329,34 +329,43 @@ func (s *state) CreateRequest(ctx context.Context) (*connector.CreateInterceptRe
+		default:
+			continue
+		}
+		if v == "" {
+			return "", errors.New("docker flag `--name` requires a value")
+		}
```
This shouldn't be `--name`, but should rather include the provided `flag`

thallgren replied:
Haha, yeah :-)

Commit:
Changes in response to code review.

- Rename getArg to getUnparsedFlagValue (with explanatory docs).
- Use flag name rather than "--name" in error string.

Signed-off-by: Thomas Hallgren <thomas@datawire.io>
